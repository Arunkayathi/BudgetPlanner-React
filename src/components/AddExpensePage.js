import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export const AddExpensePage = (props) => {
  function onSubmit(expense){
      props.onSubmit(expense);
      props.history.push('/');
  }
  return  (
  <div>
    <h1>Add Expense</h1>
    <ExpenseForm
      onSubmit={(expense)=>onSubmit(expense)}
    />
  </div>
)
};
const mapDispatchToProps=(dispatch)=>({
  onSubmit:(expense)=>dispatch(addExpense(expense))
})
export default connect(null,mapDispatchToProps)(AddExpensePage);
