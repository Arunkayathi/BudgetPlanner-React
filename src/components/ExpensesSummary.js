import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenes from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';


export const ExpensesSummary=({expenseCount,expensesTotal})=>{

    const expenseWord=expenseCount===1?'expense':'expenses';
    const formattedExpensesTotal=numeral(expensesTotal/100).format('$0,0.00');
    return (
        <div>
            {expenseCount>=1?(<h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal} </h1>):''}
        </div>
    )

};

const mapStateToProps=({expenses,filters})=>{
    const visibileExpenses=selectExpenes(expenses,filters);
    console.log(visibileExpenses);
    return {
        expenseCount:visibileExpenses.length,
        expensesTotal:selectExpensesTotal(visibileExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary)