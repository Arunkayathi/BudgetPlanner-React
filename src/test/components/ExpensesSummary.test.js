import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from './../../components/ExpensesSummary';

test("Should correctly render expenses summary with 1 expense",()=>{
    const wrapper=shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>)
    expect(wrapper).toMatchSnapshot();

});

test("Should correctly render expenses summary with multiple expenses ",()=>{
const wrapper=shallow(<ExpensesSummary expenseCount={2} expensesTotal={456}/>);
expect(wrapper).toMatchSnapshot();
});