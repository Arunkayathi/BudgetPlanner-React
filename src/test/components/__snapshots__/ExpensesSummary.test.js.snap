// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
     
  </h1>
</div>
`;

exports[`Should correctly render expenses summary with multiple expenses  1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $4.56
     
  </h1>
</div>
`;
